lib/r10k/environment/name.rb

```diff
@@ -71,8 +71,24 @@ def dirname
 
       private
 
-      def derive_prefix(source,prefix)
+      def derive_name(name, strip_component)
+        if strip_component == nil
+          name
+        elsif %r{^/.*/$}.match(strip_component)
+          regex = Regexp.new(strip_component[1..-2])
+          name.gsub(regex, '')
+        elsif name.start_with?(strip_component)
+          name[strip_component.size..-1]
+        else
+          name
+        end
+      rescue TypeError, NoMethodError
```

Comment:
IMO, it's simpler to test the class of the input then doing a rescue afterwards.

Reply:
Good call. Funny how code structure sometimes reveals developer (my) chronological thought process, and highlights afterthoughts 😅 . Pushing improvement.